### PR TITLE
add archival mock service

### LINKED ISSUE #22: Mock archiving system in a container
A complete SciCat installation should include an archiving system. This is currently treated as external to scicat by much of the documentation so it can be difficult to know how to implement it. It is also necessary for integration tests involving data transfer upon archiving and retrieval.

I suggest a minimal example container in scicatlive with the following features:

- data volumes representing 'unarchived' and 'archived' data
- protocol for copying data to and from the 'unarchived' volume (eg S3, sftp, etc)
- Basic HTTP server implementing the job API
  - Listens for job create events (probably via URL action)
  - Moves files to the 'archived' volume
  - Updates the jobStatus in scicat

### PR DESCRIPTION
**Pull request by consolethinks**
# Archival Mock-Up Service

Adds an archival mock up service that will mark datasets as archived when an archival job is requested for them. 
Fixes #22

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -58,6 +58,20 @@ services:
...
       timeout: 10s
       retries: 5
+  archive-mock:
+    build: https://github.com/SwissOpenEM/ScicatArchiveMock.git
+    depends_on:
+      rabbitmq:
+        condition: service_healthy
+      backend:
+        condition: service_started
+    environment:
```
Comment (minottic) [inline docker-compose.yaml:68]: could you use a map for the envs? just for consistency with the other service where we use environment vars

File: docker-compose.yaml [hunk 2]
```diff
@@ -58,6 +58,20 @@ services:
       interval: 30s
       timeout: 10s
       retries: 5
+  archive-mock:
+    build: https://github.com/SwissOpenEM/ScicatArchiveMock.git
+    depends_on:
+      rabbitmq:
```
Comment (minottic) [inline docker-compose.yaml:64]: just a thought. This is already covered by the dependency to the backend. I think it makes sense anyway to have it here, as it is indeed a direct dependency. 

I am happy to keep it like this if you think so

**Review by minottic — COMMENTED** [review]
very nice, thanks! Just 2 super minor comments

File: docker-compose.yaml (hunk 1 above)
Comment (consolethinks) [reply docker-compose.yaml:68]: Of course!

File: docker-compose.yaml (hunk 2 above)
Comment (consolethinks) [reply docker-compose.yaml:64]: I think I'll leave it, it represents a bit better the dependencies.

**Review by minottic — APPROVED** [review]

**Pull request merged**